private/rules/maven_publish.bzl

```diff
@@ -24,9 +25,12 @@ def _maven_publish_impl(ctx):
 
     # Expand maven coordinates for any variables to be replaced.
     coordinates = ctx.expand_make_variables("coordinates", ctx.attr.coordinates, {})
-    artifacts_short_path = ctx.file.artifact_jar.short_path if ctx.file.artifact_jar else ""
-    source_short_path = ctx.file.source_jar.short_path if ctx.file.source_jar else ""
-    javadocs_short_path = ctx.file.javadocs.short_path if ctx.file.javadocs else ""
+    artifacts_short_path = ctx.file.artifact.short_path if ctx.file.artifact else ""
+
+    classifier_artifacts_dict = {}
+    for target, classifier in ctx.attr.classifier_artifacts.items():
+        file = target.files.to_list()[0]
```

Comment:
Hmmm... this makes me nervous. Could we check that there's only one file and print a warning if there's more than one? I've a vague memory that (maybe) the kotlin rules have more than one output sometimes....

Reply:
Added some checks here.